File: implementations/springboot/examples/src/main/java/com/example/orderservice/order/domain/OrderRepository.java
package com.example.orderservice.order.domain;

import java.util.List;
import java.util.Optional;

public interface OrderRepository {
    Optional<Order> findById(String orderId);
    List<Order> findAll(OrderFindQuery query);
    long countAll(OrderFindQuery query);
    void save(Order order);
    void delete(String orderId);
}

File: implementations/kotlin-springboot/examples/src/main/kotlin/com/example/orderservice/order/domain/OrderRepository.kt
package com.example.orderservice.order.domain

interface OrderRepository {
    fun findById(orderId: String): Order?
    fun findAll(query: OrderFindQuery): List<Order>
    fun countAll(query: OrderFindQuery): Long
    fun save(order: Order)
    fun delete(orderId: String)
}

data class OrderFindQuery(
    val page: Int,
    val take: Int,
    val userId: String? = null,
    val status: List<String>? = null,
)
